### ISSUE TITLE
[0.57.4] undefined is not an object (evaluating 'warnings.filter') 

### ISSUE THREAD

**Issue by lchenfox** [facebook/react-native#28568]
Please provide all the information requested. Issues that do not follow this format are likely to stall.

## Description

Executing `npm i` and `react-native run-android` in an existing rn project(`iOS `normally), ran successfully but error produced with red screen: 

```
undefined is not an object (evaluating 'warnings.filter')
```

screenshot as follows:

![screenshot](https://user-images.githubusercontent.com/22851821/78873272-2ea09d00-7a7d-11ea-8f64-264c8041e8e7.jpg)

## React Native version:

```
  React Native Environment Info:
    System:
      OS: macOS 10.14.5
      CPU: (4) x64 Intel(R) Core(TM) i3-8100B CPU @ 3.60GHz
      Memory: 43.46 MB / 8.00 GB
      Shell: 5.3 - /bin/zsh
    Binaries:
      Node: 12.10.0 - /usr/local/bin/node
      Yarn: 1.17.3 - ~/.yarn/bin/yarn
      npm: 6.11.3 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
      Android SDK:
        API Levels: 23, 26, 27, 28
        Build Tools: 23.0.1, 26.0.1, 27.0.3, 28.0.2, 28.0.3
        System Images: android-23 | Intel x86 Atom_64, android-23 | Google APIs Intel x86 Atom, android-23 | Google APIs Intel x86 Atom_64, android-26 | Google APIs Intel x86 Atom, android-27 | Intel x86 Atom_64, android-27 | Google APIs Intel x86 Atom, android-27 | Google Play Intel x86 Atom, android-28 | Google Play Intel x86 Atom
    IDEs:
      Android Studio: 3.5 AI-191.8026.42.35.5977832
      Xcode: 11.3.1/11C505 - /usr/bin/xcodebuild
    npmPackages:
      react: ^16.6.0-alpha.8af6728 => 16.13.1
      react-native: 0.57.4 => 0.57.4
    npmGlobalPackages:
      react-native-cli: 2.0.1
      react-native-demo-for-npm: 1.0.16
      react-native-update-cli: 0.1.0
```

## Steps To Reproduce

1. Execute `npm i` in an existing project
2. run `react-native run-android`

## Expected Results

Run Android successfully without any errors.

I have alredy searched many solutions for this issue, but unfortunately, no any solutions finally. Any help for this is appreciated. Thanks a lot in advance.

**Comment by safaiyeh** [comment]
hi @lchenfox the issues here are for the latest version of react native. If you cannot upgrade, this issue is more appropriate for Stack Overflow.

Closing for now. If you can verify this on the latest version feel free to ping and I'll reopen.

**Comment by weileifrank** [comment]
> Please provide all the information requested. Issues that do not follow this format are likely to stall.
> 
> ## Description
> Executing `npm i` and `react-native run-android` in an existing rn project(`iOS `normally), ran successfully but error produced with red screen:
> 
> ```
> undefined is not an object (evaluating 'warnings.filter')
> ```
> 
> screenshot as follows:
> 
> ![screenshot](https://user-images.githubusercontent.com/22851821/78873272-2ea09d00-7a7d-11ea-8f64-264c8041e8e7.jpg)
> 
> ## React Native version:
> ```
>   React Native Environment Info:
>     System:
>       OS: macOS 10.14.5
>       CPU: (4) x64 Intel(R) Core(TM) i3-8100B CPU @ 3.60GHz
>       Memory: 43.46 MB / 8.00 GB
>       Shell: 5.3 - /bin/zsh
>     Binaries:
>       Node: 12.10.0 - /usr/local/bin/node
>       Yarn: 1.17.3 - ~/.yarn/bin/yarn
>       npm: 6.11.3 - /usr/local/bin/npm
>       Watchman: 4.9.0 - /usr/local/bin/watchman
>     SDKs:
>       iOS SDK:
>         Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
>       Android SDK:
>         API Levels: 23, 26, 27, 28
>         Build Tools: 23.0.1, 26.0.1, 27.0.3, 28.0.2, 28.0.3
>         System Images: android-23 | Intel x86 Atom_64, android-23 | Google APIs Intel x86 Atom, android-23 | Google APIs Intel x86 Atom_64, android-26 | Google APIs Intel x86 Atom, android-27 | Intel x86 Atom_64, android-27 | Google APIs Intel x86 Atom, android-27 | Google Play Intel x86 Atom, android-28 | Google Play Intel x86 Atom
>     IDEs:
>       Android Studio: 3.5 AI-191.8026.42.35.5977832
>       Xcode: 11.3.1/11C505 - /usr/bin/xcodebuild
>     npmPackages:
>       react: ^16.6.0-alpha.8af6728 => 16.13.1
>       react-native: 0.57.4 => 0.57.4
>     npmGlobalPackages:
>       react-native-cli: 2.0.1
>       react-native-demo-for-npm: 1.0.16
>       react-native-update-cli: 0.1.0
> ```
> 
> ## Steps To Reproduce
> 1. Execute `npm i` in an existing project
> 2. run `react-native run-android`
> 
> ## Expected Results
> Run Android successfully without any errors.
> 
> I have alredy searched many solutions for this issue, but unfortunately, no any solutions finally. Any help for this is appreciated. Thanks a lot in advance.

I get same error. do you resolve it?

**Comment by liyang521** [comment]
I also met this problem, how to solve excuse me, thank you very much

**Comment by lchenfox** [comment]
> I also met this problem, how to solve excuse me, thank you very much

I had not found the solution about this issue until three days ago. So I just debug via running iOS simulator all the time. But about three days ago, I initialized a new project using `react-native init demo --version 0.57.4`. Then copy `devDependencies` in `package.json` as follows:

```
 "devDependencies": {
    "babel-jest": "26.0.1",
    "jest": "26.0.1",
    "metro-react-native-babel-preset": "0.59.0",
    "react-test-renderer": "16.6.0-alpha.8af6728"
  },
```

into `package.json` in old project(my project). Then 

1. `npm start -- --reset-cache`
2. `rm -rf node_modules`
3. `npm install`
4. delete `app` on emulator on Android
5.  Restart the computer

At last, the error gent away and `console.warn()` performs normally on Android without Red Screen as can be seen above. I guess it causes this issue because of `npm i` so `devDependencies` was changed. Hope it works for you. Good luck.
